### ISSUE TITLE
With scaling=1 there shouldn't be numbered containers

### ISSUE THREAD

**Issue by cybe** [byrnedo/capitan#1]
The latest version introduced scaling. As I'm not using that particular feature, I'm left with all my containers being named as "${container}1" with the "1" being appended unnecessarily. This also broke some of my internal DNS references between containers, as the container name equals the hostname. I therefore propose to omit the number in case of scaling=1.

**Comment by byrnedo** [comment]
Hi @cybe,

First I'm glad you're using Capitan :)
I get what you mean. I was mainly going with convention when it came to this and I actually wasn't really aware of who all was using Capitan so I'd been committing quite a lot of code in the last while.

That said,  I'm a bit reluctant to change it back as I think it's less ambiguous the way it is now, can I maybe first help you troubleshoot you setup and see is there another way round it?

Edit: I've also created a [releases](https://github.com/byrnedo/capitan/releases) page now and will be doing proper versioning from now on.

**Comment by byrnedo** [comment]
@cybe bump

**Comment by cybe** [comment]
Hey Donal,

thanks for the bump, somehow missed answering you. First of, capitan is doing well in my setup, since I adapted to the new naming scheme. Already did a few "up"s in production that ran as expected. As a side note though, I'm only running a small setup with about 10 containers on a single host for some private services. Also without any load-balancing or fail-overs, hence there is no need for me to have multiple containers per image. You can take a look at my capitan-configs here: https://github.com/cybe/docker-capitan/tree/master/stacks

Anyway, back to topic: I agree with you to favor consistency over some rather insignificant naming scheme. But I can still see the need for configurable container names to achieve     some sort of predictability in terms of the hostname. Given the nature of scaling, providing a configurable distinct hostname adds some complexity and might break consistency. Haven't thought that through.

How do you manage container accessing each other? The container in my setup uses hard coded hostnames for this purpose that are built into the image.

P.S.: I didn't mind the breaking changes, since this is a young project and I was using the master branch anyway. But thank you for providing releases, I will switch to those eventually.

**Comment by byrnedo** [comment]
Glad to hear it's doing well. Well, regarding hostnames, you now have
access to a few env's when capitan evaluates the run command, check out
https://github.com/byrnedo/capitan#environment-variables.
So what I've been doing is just:

`mysql hostname \$CAPITAN_CONTAINER_NAME`

And then the hostname matches the container name.

I'm using consul to do the dns with the main target of a multi-host
docker-swarm:
https://github.com/byrnedo/dev-stack/blob/master/discovery/capitan.cfg.sh
Basically if I'm only running on one host it just makes a one node consul
cluster (bad I know, but I'm just using this for when I don't want to spin
up vms). Otherwise on a multi-node cluster it expects an existing consul
cluster to exist and just acts as an agent for that cluster.

**Comment by cybe** [comment]
Using the those environment variables could work while doing some extra work from within the container. One have to access the environment variables from within and set up the /etc/hosts file accordingly.

I actually just discovered that the hostname provided via the docker run command would indeed set that hostname to /etc/hostname and /etc/hosts (as expected). But that now running container is reachable from other containers in the same network via its container name instead of the configured hostname. I'm a bit confused by that particular behavior of docker. The automatically generated /etc/hosts seems to be always based on container names. I'll look into that.

Using something like consul or any other dedicated discovery mechanism seems a bit overkill for my situation. But I might look into that option as well. Thanks for your example.

I guess you can close this issue now, since my reported issue turned out to be the expected behavior.

**Comment by byrnedo** [comment]
Yeah, I noticed that too with the new networking that the container name is what gets put into /etc/hosts.

I have to say consul was doing fine with just one node, the bonus is that if I bind consul's dns port to the docker bridge, and then add the bridge ip to my own laptops dns servers, I can then access all the containers by their dns name. 

I'll close the issue but thanks for your interest in Capitan and please let me know if you have any ideas about additions or changes.
